api/ws/rpc/connect.js

```diff
@@ -199,8 +199,14 @@ const connectSteps = {
 					peer.ip
 				} failed with code ${code} and reason - ${reason}`
 			);
-			socket.destroy();
-			onDisconnectCb();
+
+			if (peer.socket && peer.socket.state === peer.socket.CLOSED) {
```

Comment:
is it `peer.socket.state === peer.socket.CLOSED` or `peer.socket.state === Peer.STATE.CLOSED`?

Reply:
@ManuGowda it's peer.socket.state not peer.state because we want to rely on the socket's state not the peer's state.